src/engine/strat_engine/backstore/setup.rs

```diff
@@ -109,17 +108,12 @@ pub fn get_metadata(
                 None
             }
         })
-        .next();
-
-    if let Some(psave) = poolsave {
-        return Ok(Some((*most_recent_time, psave)));
-    }
-
-    // If no data has yet returned, we have an error. That is, we should have
-    // some metadata, because we have a most recent time, but we failed to
-    // get any.
-    let err_str = "timestamp indicates data was written, but no data successfully read";
-    Err(StratisError::Engine(ErrorEnum::NotFound, err_str.into()))
+        .next()
+        .ok_or(StratisError::Engine(
```

Comment:
clippy, quite correctly, would like to see an ```ok_or_else``` here. I agree, otherwise looks fine to me.